lib/engine/game/g_18_mex/step/buy_sell_par_shares.rb

```diff
@@ -9,15 +9,24 @@ module G18MEX
       module Step
         class BuySellParShares < Engine::Step::BuySellParShares
           def can_buy?(entity, bundle)
-            super && !attempt_ndm_action_on_unavailable?(bundle)
+            return false if attempt_ndm_action_on_unavailable?(bundle)
+
+            super
           end
 
           def can_sell?(entity, bundle)
-            super && !attempt_ndm_action_on_unavailable?(bundle)
+            return false if attempt_ndm_action_on_unavailable?(bundle)
+
+            super
           end
 
-          def can_gain?(entity, bundle)
-            super && !attempt_ndm_action_on_unavailable?(bundle)
+          def can_gain?(entity, bundle, exchange: false)
+            return false if attempt_ndm_action_on_unavailable?(bundle)
+            return if !bundle || !entity
```

Comment:
The `!bundle` check should go first, right?

Reply:
AFAICT, everything after the `attempt_ndm_action_on_unavailable?` line is based on [`Engine::Step::ShareBuying#can_gain?`](https://github.com/tobymao/18xx/blob/master/lib/engine/step/share_buying.rb#L36-L43), with a change to the last line.

So having the `!bundle` check second is I think is basically the same as making it `!attempt_ndm_action_on_unavailable?(bundle) && super`

But on second glance, it does seem like if `bundle` is `nil`, putting the `!bundle` check first would catch it, but with the `attempt_ndm_action_on_unavailable?(bundle)` call first, it is prone to a "nil pointer" error.

Reply:
Changed to make it bundle, entity, ndm-unavailable

It is correct that new code is based on super method. Needed to copy it to change the cert limit check. And I then removed one check which I think did not apply to 18MEX.